### DISCUSSION TITLE
Noise from Language Server

### DISCUSSION THREAD

**Discussion by soeguet** [nvim-java/nvim-java#89] [Q&A] [answered]
hey there, does anybody happen to know how to get rid of the noisy messages from the lsp:
![Screenshot_20231217_162247](https://github.com/nvim-java/nvim-java/assets/117000423/5eedb865-c05f-45a4-96e5-85daefeb6ca7)

overriding the handler worked with nvim-jdtls like so:
```
    handlers = {
        ['language/status'] = function(_, result)
            -- Print or whatever.
        end,
        ['$/progress'] = function(_, result, ctx)
            -- disable progress updates.
        end,
    },
```

**Comment by s1n7ax** [comment, marked as answer]
@soeguet 
Hi, those are setup configuration of language servers. Those should be passed to the `lspconfig.jdtls.setup` function. However there was a bug, I already fixed it so,

1. Update the plugins
2. Pass the setup config

```lua
require('lspconfig').jdtls.setup({
			handlers = {
				['language/status'] = function(_, result)
					vim.print('***')
				end,
				['$/progress'] = function(_, result, ctx)
					vim.print('---')
				end,
			},
		})
```

**Reply by s1n7ax** [reply]
You get those notification from mainly `$/progress` handler. You can override that with an empty function to get rid of them

**Reply by soeguet** [reply]
updating did the trick, thanks!
